src/DynamoDbQueryBuilder.php

```diff
@@ -293,10 +298,31 @@ protected function getAll($columns = [], $limit = -1, $use_iterator = true)
         return new Collection($results);
     }
 
+    protected function conditionsAreExactSearch()
+    {
+        if (empty($this->where)) {
+            return false;
+        }
+
+        foreach ($this->where as $condition) {
+            if (array_get($condition, 'ComparisonOperator') !== ComparisonOperator::EQ) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
     /**
      * Check if conditions "where" contain primary key or composite key.
      * For composite key, it will return false if the conditions don't have all composite key.
      *
+     * For example:
+     *   Consider a composite key condition:
+     *     $model->where('partition_key', 'foo')->where('sort_key', 'bar')
+     *   We return ['partition_key' => 'foo', 'sort_key' => 'bar'] since the conditions
+     *   contain all the composite key.
+     *
      * @return array|bool the condition value
      */
     protected function conditionsContainKey()
```

Comment:
Maybe should make this more strict. If the conditions have both key and others, return false.